frontends/p4/simplifyParsers.cpp

```diff
@@ -46,6 +46,20 @@ class RemoveUnreachableStates : public Transform {
             transitions->reachable(start->to<IR::ParserState>(), reachable);
             // Remove unreachable states from call-graph
             transitions->restrict(reachable);
+            // If neither the accept nor the reject states are
+            // reachable the parser we signal an error.  The reject
+            // state will probably be reached through an error.
+            bool acceptReachable = false;
+            bool rejectReachable = false;
+            for (auto s : reachable) {
+                if (s->name == IR::ParserState::reject)
+                    rejectReachable = true;
+                else if (s->name == IR::ParserState::accept)
+                    acceptReachable = true;
+            }
+            if (!rejectReachable && !acceptReachable)
```

Comment:
Should it always be an error (or at least a warning?) if the accept state is unreachable, regardless of the reachability of reject?  I guess it could make sense to have a sub-parser that always rejects and is called in the event of an error from another parser?

Reply:
Not reaching accept was already a warning.
Now we added an error if the parser is definitely in an infinite loop.
I did this because otherwise the dataflow analysis finds no definitions for variables used in the unreachable states and it crashes.